Packs/SailPointIdentityNow/Integrations/SailPointIdentityNow/SailPointIdentityNow.py

```diff
@@ -0,0 +1,622 @@
+from CommonServerPython import *
+
+''' IMPORTS '''
+
+import traceback
+
+import requests
+import urllib3
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+''' CONSTANTS '''
+
+# IdentityNow OAuth token endpoint
+IDN_OAUTH_EXT = '/oauth/token'
+IDN_SEARCH_PREFIX = '/beta/search'
+
+# Resource endpoints
+IDN_SEARCH_IDENTITIES_EXT = f'{IDN_SEARCH_PREFIX}/identities'
+IDN_BETA_ACCOUNTS_EXT = '/beta/accounts'
+IDN_SEARCH_ACCESS_PROFILES_EXT = f'{IDN_SEARCH_PREFIX}/accessprofiles'
+IDN_SEARCH_ROLES_EXT = f'{IDN_SEARCH_PREFIX}/roles'
+IDN_SEARCH_ENTITLEMENTS_EXT = f'{IDN_SEARCH_PREFIX}/entitlements'
+IDN_SEARCH_EVENTS_EXT = f'{IDN_SEARCH_PREFIX}/events'
+IDN_BETA_ACCOUNT_ACTIVITIES_EXT = '/beta/account-activities'
+IDN_BETA_ACCESS_REQUEST_EXT = '/beta/access-requests'
+
+# Other constants
+# Using beta account API for accounts
+SEARCHABLE_TYPE_LIST = ['identities', 'accessprofiles', 'roles', 'entitlements', 'events']
+MAX_INCIDENTS_TO_FETCH = 250
+MAX_LIMIT = 250
+OFFSET_DEFAULT = '0'
+LIMIT_DEFAULT = '250'
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%S'
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """
+    Client class to interact with API request.
+    """
+
+    def __init__(self, base_url: str, verify: bool, proxy: bool, headers: dict, max_results: int, request_timeout: int):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, headers=headers)
+        self.max_results = max_results
+        self.request_timeout = request_timeout
+
+    def send_request(self, url_suffix: str, method: str, params=None, json_data=None):
+        """
+        Perform a HTTP request to IdentityNow API.
+
+        :type url_suffix: ``str``
+        :param url_suffix: IdentityNow API endpoint ext suffix.
+
+        :type method: ``str``
+        :param method: HTTP method, e.g. 'GET', 'POST', 'PUT', 'DELETE'.
+
+        :type params: ``JSON``
+        :param params: URL parameters to specify the query.
+
+        :type json_data: ``JSON``
+        :param json_data: Data to be sent as part of 'POST' or 'PUT' request.
+
+        :return: Response after fulfilling the request successfully, else None.
+        """
+        if url_suffix is None or method is None:
+            return None
+
+        return self._http_request(url_suffix=url_suffix, method=method, json_data=json_data, params=params,
+                                  timeout=self.request_timeout, resp_type='response',
+                                  ok_codes=(200, 201, 202, 204, 400, 401, 403, 404, 429, 500), proxies=handle_proxy())
```

Comment:
1. Yes, I am sure about the codes. Is there any incentive of letting the http_request handle it?
2. The passing of proxy is done in case the partner wants to have a different connectivity setup from the convention one. If there is a suggestion here of not passing the proxy settings, please let me know.

Reply:
1. Its just that we are dealing with the request codes in the http_request method: 
![image](https://user-images.githubusercontent.com/50294648/110610308-c54a7200-8196-11eb-8023-c6a7b961b045.png)

So in case of an error code (403,404,500) it will be handled there.
2. So since you are passing the proxy argument to the base client constructor, it already takes care of the proxy settings so it is not needed to pass it to the request.

Reply:
Updated.